java/src/main/java/ai/onnxruntime/OrtSession.java

```diff
@@ -1003,6 +1003,16 @@ public void addCoreML(EnumSet<CoreMLFlags> flags) throws OrtException {
     /**
      * Adds Xnnpack as an execution backend. Needs to list all options here if a new option
      * supported. current supported options: {}
+     * The maximum number of provider options is set to 128 (see addExecutionProvider's comment
+     * or ONNXRuntime PR #14076 for the reason). This number is controlled by
+     * ORT_JAVA_MAX_ARGUMENT_ARRAY_LENGTH in ai_onnxruntime_OrtSession_SessionOptions.c. If 128
```

Comment:
Do you want me to work up a PR which does this enforcement? I think we probably want a private method that all the `addExecutionProvider` calls go through to enforce it, then refactor `addXnnpack` to call that instead of the native method directly.

Reply:
I am open for discussion. It seems the earlier we bail and the earlier we see the constant the better. And both would be true in Java.

Reply:
Ok, I'll find some time in a few weeks to put a PR together.